test/integration/filters/metadata_stop_all_filter.cc

```diff
@@ -0,0 +1,74 @@
+#include <chrono>
+#include <string>
+
+#include "envoy/event/timer.h"
+#include "envoy/http/filter.h"
+#include "envoy/registry/registry.h"
+#include "envoy/server/filter_config.h"
+
+#include "common/buffer/buffer_impl.h"
+
+#include "extensions/filters/http/common/empty_http_filter_config.h"
+#include "extensions/filters/http/common/pass_through_filter.h"
+
+#include "test/integration/filters/common.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+
+class MetadataStopAllFilter : public Http::PassThroughFilter {
```

Comment:
This filter does a lot of similar thing as DecodeHeadersReturnStopAllFilter. Can we extend that one and re-use it instead?

Reply:
I think separating them may make it easier to read the test. DecodeHeadersReturnStopAllFilter is already messy with many special code to verify data StopAll functions. May I keep the current file?